File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/BuyerAuthResponse.cs
#region "Copyright"

//PayPal Payflow Pro .NET SDK
//Copyright (C) 2014  PayPal, Inc.
//
//This file is part of the Payflow Pro .NET SDK
//
//The Payflow .NET SDK is free software: you can redistribute it and/or modify
//it under the terms of the GNU General Public License as published by
//the Free Software Foundation, either version 3 of the License, or
//any later version.
//
//The Payflow .NET SDK is is distributed in the hope that it will be useful,
//but WITHOUT ANY WARRANTY; without even the implied warranty of
//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//GNU General Public License for more details.
//
//You should have received a copy of the GNU General Public License
//along with the Payflow .NET SDK.  If not, see <http://www.gnu.org/licenses/>.


#endregion

#region "Imports"

using System;
using System.Collections;
using PayPal.Payments.Common.Utility;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for the Buyer Authorization operation
	/// </summary>
	public class BuyerAuthResponse : BaseResponseDataObject
	{
		#region "CONSTRUCTOR"

		internal BuyerAuthResponse()
		{}

		#endregion

		#region "Member Variables"

		/// <summary>
		/// Holds Acs Url
		/// </summary>
		private String mAcsUrl;

		/// <summary>
		/// Holds Authentication Id
		/// </summary>
		private String mAuthenticationId;

		/// <summary>
		/// Holds Authentication status
		/// </summary>
		private String mAuthenticationStatus;

		/// <summary>
		/// Holds CAVV
		/// </summary>
		private String mCavv;

		/// <summary>
		/// Holds ECI
		/// </summary>
		private String mEci;

		/// <summary>
		/// Holds PAREQ
		/// </summary>
		private String mPaReq;

		/// <summary>
		/// Holds XID
		/// </summary>
		private String mXid;

		/// <summary>
		/// Holds MD
		/// </summary>
		private String mMd;

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets the AcsUrl parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>ACSURL</code>
		/// </remarks>
		public String AcsUrl
		{
			get { return mAcsUrl; }
		}

		/// <summary>
		/// Gets the authentication_id parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>AUTHENTICATION_ID</code>
		/// </remarks>
		public String Authentication_Id
		{
			get { return mAuthenticationId; }
		}

		/// <summary>
		/// Gets the authentication_status parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>AUTHENTICATION_STATUS</code>
		/// </remarks>
		public String Authentication_Status
		{
			get { return mAuthenticationStatus; }
		}

		/// <summary>
		/// Gets the CAVV parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>CAVV</code>
		/// </remarks>
		public String CAVV
		{
			get { return mCavv; }
		}

		/// <summary>
		/// Gets the ECI parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>ECI</code>
		/// </remarks>
		public String ECI
		{
			get { return mEci; }
		}

		/// <summary>
		/// Gets the PaReq parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PAREQ</code>
		/// </remarks>
		public String PaReq
		{
			get { return mPaReq; }
		}

		/// <summary>
		/// Gets the XID parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>XID</code>
		/// </remarks>
		public String XID
		{
			get { return mXid; }
		}

		/// <summary>
		/// Gets the MD parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>MD</code>
		/// </remarks>
		public String MD
		{
			get { return mMd; }
		}

		#endregion

		#region "Functions"

		/// <summary>
		/// Sets the Response parameters in Response data objects.
		/// </summary>
		///<param name="ResponseHashTable">Response Hash table by ref</param>
		internal void SetParams(ref Hashtable ResponseHashTable)
		{
			mAcsUrl = (String) ResponseHashTable[PayflowConstants.PARAM_ACSURL];
			mAuthenticationId = (String) ResponseHashTable[PayflowConstants.PARAM_AUTHENICATION_ID];
			mAuthenticationStatus = (String) ResponseHashTable[PayflowConstants.PARAM_AUTHENICATION_STATUS];
			mCavv = (String) ResponseHashTable[PayflowConstants.PARAM_CAVV];
			mEci = (String) ResponseHashTable[PayflowConstants.PARAM_ECI];
			mMd = (String) ResponseHashTable[PayflowConstants.PARAM_MD];
			mPaReq = (String) ResponseHashTable[PayflowConstants.PARAM_PAREQ];
			mXid = (String) ResponseHashTable[PayflowConstants.PARAM_XID];

			ResponseHashTable.Remove(PayflowConstants.PARAM_ACSURL);
			ResponseHashTable.Remove(PayflowConstants.PARAM_AUTHENICATION_ID);
			ResponseHashTable.Remove(PayflowConstants.PARAM_AUTHENICATION_STATUS);
			ResponseHashTable.Remove(PayflowConstants.PARAM_CAVV);
			ResponseHashTable.Remove(PayflowConstants.PARAM_ECI);
			ResponseHashTable.Remove(PayflowConstants.PARAM_MD);
			ResponseHashTable.Remove(PayflowConstants.PARAM_PAREQ);
			ResponseHashTable.Remove(PayflowConstants.PARAM_XID);
		}

		#endregion
	}

}

File: java/src/sdk/dataobjects/paypal/payflow/BuyerAuthResponse.java
package paypal.payflow;

import java.util.Hashtable;

/**
 * Used for the buyerauth operation
 */
public class BuyerAuthResponse extends BaseResponseDataObject {


    private String acsUrl;
    private String authenticationId;
    private String authenticationStatus;
    private String cavv;
    private String eci;
    private String paReq;
    private String xid;
    private String md;

    protected BuyerAuthResponse() {
    }

    /**
     * Gets the acsurl parameter.
     *
     * @return String
     * @paypal.sample <p>Maps to Payflow Parameter: ACSURL</p>
     */
    public String getAcsUrl() {
        return acsUrl;
    }

    /**
     * Sets the acsurl parameter.
     *
     * @param acsUrl String
     * @paypal.sample <p>Maps to Payflow Parameter: ACSURL</p>
     */
    public void setAcsUrl(String acsUrl) {
        this.acsUrl = acsUrl;
    }

    /**
     * Gets the authentication_id parameter.
     *
     * @return authenticationId String
     * @paypal.sample <p>Maps to Payflow Parameter: AUTHENTICATION_ID</p>
     */
    public String getAuthenticationId() {
        return authenticationId;
    }

    /**
     * Sets the authentication_id parameter.
     *
     * @param authenticationId String
     * @paypal.sample <p>Maps to Payflow Parameter: AUTHENTICATION_ID</p>
     */
    public void setAuthenticationId(String authenticationId) {
        this.authenticationId = authenticationId;
    }

    /**
     * Gets the authentication_status parameter.
     *
     * @return authenticationStatus String
     * @paypal.sample <p>Maps to Payflow Parameter: AUTHENTICATION_STATUS</p>
     */
    public String getAuthenticationStatus() {
        return authenticationStatus;
    }

    /**
     * Sets the authentication_status parameter.
     *
     * @param authenticationStatus String
     * @paypal.sample <p>Maps to Payflow Parameter: AUTHENTICATION_STATUS</p>
     */
    public void setAuthenticationStatus(String authenticationStatus) {
        this.authenticationStatus = authenticationStatus;
    }

    /**
     * Gets the CAVV parameter.
     *
     * @return cavv String
     * @paypal.sample <p>Maps to Payflow Parameter: CAVV</p>
     */
    public String getCavv() {
        return cavv;
    }

    /**
     * Sets the CAVV parameter.
     *
     * @param cavv String
     * @paypal.sample <p>Maps to Payflow Parameter: CAVV</p>
     */
    public void setCavv(String cavv) {
        this.cavv = cavv;
    }

    /**
     * Gets the ECI parameter.
     *
     * @return eci String
     * @paypal.sample <p>Maps to Payflow Parameter: ECI</p>
     */
    public String getEci() {
        return eci;
    }

    /**
     * Sets the ECI parameter.
     *
     * @param eci String
     * @paypal.sample <p>Maps to Payflow Parameter: ECI</p>
     */
    public void setEci(String eci) {
        this.eci = eci;
    }

    /**
     * Gets the MD parameter.
     *
     * @return md String
     * @paypal.sample <p>Maps to Payflow Parameter: MD</p>
     */
    public String getMd() {
        return md;
    }

    /**
     * Sets the MD parameter.
     *
     * @param md String
     * @paypal.sample <p>Maps to Payflow Parameter: MD</p>
     */
    public void setMd(String md) {
        this.md = md;
    }

    /**
     * Gets the PaReq parameter.
     *
     * @return paReq String
     * @paypal.sample <p>Maps to Payflow Parameter: PAREQ</p>
     */
    public String getPaReq() {
        return paReq;
    }

    /**
     * Sets the PaReq parameter.
     *
     * @param paReq String
     * @paypal.sample <p>Maps to Payflow Parameter: PAREQ</p>
     */
    public void setPaReq(String paReq) {
        this.paReq = paReq;
    }

    /**
     * Gets the XID parameter.
     *
     * @return xid String
     * @paypal.sample <p>Maps to Payflow Parameter: XID</p>
     */
    public String getXid() {
        return xid;
    }

    /**
     * Sets the XID parameter.
     *
     * @param xid String
     * @paypal.sample <p>Maps to Payflow Parameter: XID</p>
     */
    public void setXid(String xid) {
        this.xid = xid;
    }

    protected void setParams(Hashtable ResponseHashTable) {

        acsUrl = (String) ResponseHashTable.get(PayflowConstants.PARAM_ACSURL);
        authenticationId = (String) ResponseHashTable.get(PayflowConstants.PARAM_AUTHENTICATION_ID);
        authenticationStatus = (String) ResponseHashTable.get(PayflowConstants.PARAM_AUTHENICATION_STATUS);
        cavv = (String) ResponseHashTable.get(PayflowConstants.PARAM_CAVV);
        eci = (String) ResponseHashTable.get(PayflowConstants.PARAM_ECI);
        md = (String) ResponseHashTable.get(PayflowConstants.PARAM_MD);
        paReq = (String) ResponseHashTable.get(PayflowConstants.PARAM_PAREQ);
        xid = (String) ResponseHashTable.get(PayflowConstants.PARAM_XID);
        ResponseHashTable.remove(PayflowConstants.PARAM_ACSURL);
        ResponseHashTable.remove(PayflowConstants.PARAM_AUTHENTICATION_ID);
        ResponseHashTable.remove(PayflowConstants.PARAM_AUTHENICATION_STATUS);
        ResponseHashTable.remove(PayflowConstants.PARAM_CAVV);
        ResponseHashTable.remove(PayflowConstants.PARAM_ECI);
        ResponseHashTable.remove(PayflowConstants.PARAM_MD);
        ResponseHashTable.remove(PayflowConstants.PARAM_PAREQ);
        ResponseHashTable.remove(PayflowConstants.PARAM_XID);
    }

}